pkg/operators/v1alpha1/subscription_types.go

```diff
@@ -84,6 +84,18 @@ type SubscriptionConfig struct {
 	// List of VolumeMounts to set in the container.
 	// +optional
 	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+
+	// Describes node affinity scheduling rules for the pod.
+	// +optional
+	NodeAffinity *corev1.NodeAffinity `json:"nodeAffinity,omitempty" protobuf:"bytes,1,opt,name=nodeAffinity"`
+
+	// Describes pod affinity scheduling rules (e.g. co-locate this pod in the same node, zone, etc. as some other pod(s)).
+	// +optional
+	PodAffinity *corev1.PodAffinity `json:"podAffinity,omitempty" protobuf:"bytes,2,opt,name=podAffinity"`
+
+	// Describes pod anti-affinity scheduling rules (e.g. avoid putting this pod in the same node, zone, etc. as some other pod(s)).
+	// +optional
+	PodAntiAffinity *corev1.PodAntiAffinity `json:"podAntiAffinity,omitempty" protobuf:"bytes,3,opt,name=podAntiAffinity"`
```

Comment:
Are there tradeoffs between explicitly including these three fields in the `spec.config` vs including the top level `corev1.Affinity` (which contains all of these)?

Reply:
it would give you more granular control without having to reproduce what's already defined in the deployment pod spec.
For instance, the stock deployment might define all three, but you might only want to change one. If we were to use just affinity, the UX becomes a bit wonky, imo. e.g. 

```yaml
affinity:
  nodeAffinity: ...
```

Does this mean, _only_ override nodeAffinity and keep the others stock. Or, does it mean override the whole `affinity` stanza removing the others?

I thought by splitting them up it would be clearer what is actually being overridden and what is not. wdyt? @joelanford

Reply:
That's what I figured you'd say. My only counter argument is: if a new `affinity` sub-field is added in the future, we will have to add explicit support for it (i.e. more than just vendoring the new API version).

Reply:
That's true. But I'd still fall on the UX side rather than DX side. Let me check what the other configs do. If we can get away with saying is e.g. affinity.podAffinity = nil -> don't override affinity.podAffinity. But if they want to nil it out, it would need to be:

```yaml
affiinity:
  nodeAffinity: ..
  podAffinity: {}
```

(I'm assuming here the meaning of default podAffinity == nil - though if this changes, it could cause trouble)

Reply:
> (I'm assuming here the meaning of default podAffinity == nil - though if this changes, it could cause trouble)

Hmmm...probably can't get away from this {} could be different than nil problem though.

Reply:
I don't have a strong opinion really. Just wanted to make sure it was considered.

Reply:
If you're good with it, then give us a lgtm =D
tbh it's probably not a huge risk either way, once OLM 1.x comes in to action this will go away...

Reply:
> tbh it's probably not a huge risk either way, once OLM 1.x comes in to action this will go away...

I was still on the fence looking at these changes earlier, and whether this was the right UX going forward. But I'm also fine with this perspective, and we just need a short term bandaid while we continue to iterate on these new OLM 1.x components and behaviors.

/lgtm
/hold